scripts/generate/index.ts

```diff
@@ -1,10 +1,52 @@
 import yargs from 'yargs';
-import {generateSrc, Template} from './generate-src';
+import {
+  log,
+  generateSrc,
+  Template,
+  EXTENSION_TEMPLATE_MAP,
+} from './generate-src';
 import {cleanUp} from './clean-up';
 
 // eslint-disable-next-line @typescript-eslint/no-var-requires
 const inquirer = require('inquirer');
 
+const NICE_TEMPLATE_NAME_MAP = new Map([
+  [Template.Vanilla, 'vanilla JavaScript'],
+  [Template.React, 'React'],
+  [Template.VanillaTypescript, 'TypeScript'],
+  [Template.ReactTypescript, 'React and TypeScript'],
+]);
+
+(async () => {
+  const {type: extensionType, template: templateIdentifier} = yargs.argv;
+
+  const type = validateExtensionType(extensionType);
```

Comment:
In general this case is not encountered because the Shopify CLI provides a picker for the type if you don't declare one. Agreed that this is not well-structured and there are tons of edges that will error badly, see earlier comment about just wanting something that works :P